src/Microsoft.AspNet.Razor/Generator/Compiler/CodeBuilder/CSharp/CSharpTagHelperCodeRenderer.cs

```diff
@@ -75,15 +75,8 @@ public void RenderTagHelper(TagHelperChunk chunk)
             if (!_designTimeMode)
             {
                 RenderRunTagHelpers();
-
-                RenderTagOutput(_tagHelperContext.OutputGenerateStartTagMethodName, methodReturnsString: true);
-                RenderTagOutput(_tagHelperContext.OutputGeneratePreContentMethodName, methodReturnsString: false);
-
-                RenderTagHelperContent();
-
-                RenderTagOutput(_tagHelperContext.OutputGeneratePostContentMethodName, methodReturnsString: false);
-                RenderTagOutput(_tagHelperContext.OutputGenerateEndTagMethodName, methodReturnsString: true);
-
+                RenderStartWriteTagHelperMethod();
+                _writer.Write(ExecutionContextVariableName).WriteEndMethodInvocation();
```

Comment:
nit: couldn't this be done inside `RenderStartWriteTagHelperMethod()`, perhaps after renaming that method to (say) `RenderWriteTagHelperMethodCall`?